src/main/java/ch/njol/skript/util/EnumUtils.java

```diff
@@ -43,55 +40,67 @@ public EnumUtils(final Class<E> c, final String languageNode) {
 		
 		this.c = c;
 		this.languageNode = languageNode;
-		
+
 		names = new String[c.getEnumConstants().length];
 		
-		Language.addListener(new LanguageChangeListener() {
-			@Override
-			public void onLanguageChange() {
-				validate(true);
-			}
-		});
+		Language.addListener(() -> validate(true));
 	}
 	
 	/**
 	 * Updates the names if the language has changed or the enum was modified (using reflection).
 	 */
-	final void validate(final boolean force) {
+	void validate(final boolean force) {
 		boolean update = force;
-		
-		final int newL = c.getEnumConstants().length;
-		if (newL > names.length) {
-			names = new String[newL];
+
+		int newLength = c.getEnumConstants().length;
+		if (newLength != names.length) {
+			names = new String[newLength];
 			update = true;
 		}
-		
+		for (E constant : c.getEnumConstants()) {
```

Comment:
I wonder whether there might be a neat way to do the should-update checks together?